sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -1228,7 +1228,7 @@ colPosition
     ;
 
 collateClause
-    : COLLATE collationName=identifier
+    : COLLATE collationName=qualifiedName
```

Comment:
shall we use `multipartIdentifier`? Which provides better error message for some common wrong identifiers.